torch/lib/THS/generic/THSTensorMath.c

```diff
@@ -15,6 +15,18 @@ void THSTensor_(zero)(THSTensor *self) {
   self->nnz = 0;
 }
 
+void THSTensor_(zeros)(THSTensor *r_, THLongStorage *size)
+{
+  THSTensor_(resize)(r_, size);
```

Comment:
Doesn't sparse resize fix indices or sth? It seems a more expensive call than for dense resize. Maybe zero it beforehand (zero is cheap in sparse)

Reply:
Actually sparse resize doesn't fix indices. It simply sets the sizes. cpu:https://github.com/pytorch/pytorch/blob/master/torch/lib/THS/generic/THSTensor.c#L266 (same for gpu)